core/src/main/java/jenkins/security/ResourceDomainConfiguration.java

```diff
@@ -168,8 +170,11 @@ private FormValidation checkUrl(String resourceRootUrlString, boolean allowOnlin
                 }
                 // response is error
                 String responseMessage = httpURLConnection.getResponseMessage();
-                if (responseCode == 404 && responseMessage.equals(ERROR_RESPONSE)) {
-                    return FormValidation.ok(Messages.ResourceDomainConfiguration_ResourceResponse());
+                if (responseCode == 404) {
+                    String responseBody = String.join("", IOUtils.readLines(httpURLConnection.getErrorStream(), StandardCharsets.UTF_8));
```

Comment:
`StandardCharsets.UTF_8` is probably wrong but unless it's likely to throw exceptions I doubt I care enough.